api/src/main/java/org/hawkular/inventory/api/filters/Filter.java

```diff
@@ -34,7 +34,15 @@
  * A base class for filters. Defines no filtering logic in and of itself.
  *
  * <p>The implementations of the Hawkular inventory API are supposed to support filtering by {@link Related},
- * {@link With.Ids} and {@link With.Types}.
+ * {@link With.Ids} and {@link With.Types}. There is also a sub-class of filters for the relation filtering {@link
+ * RelationFilter}.
+ *
+ * To create these filters, feel free to use the static helper methods defined on {@link With}.
+ * <p>
+ * Note: Additional information for the library consumers.
+ * Don't extend this class with the hope the new filter will work without any further work, It needs to be registered
```

Comment:
This leaves hope that after "registration" it might work. I'd use something stronger as:
"Don't extend this class with hope that the new filter will work. This class is extendable only for the benefit of the API implementations that can reuse it internally. For the users of the API, only the subclasses of Filter declared directly in the API are available."